pageserver/src/tenant/timeline.rs

```diff
@@ -3171,6 +3266,7 @@ impl Timeline {
         request_lsn: Lsn,
         reconstruct_state: &mut ValuesReconstructState,
         ctx: &RequestContext,
+        is_metadata_scan: bool,
```

Comment:
I don't like this bimodal behaviour. Let's update the logic on line 3297 to make it aware of your new non inheritable range (`METADATA_KEY_BEGIN_PREFIX..METADATA_KEY_END_PREFIX`).

The issue you need to deal with is that we currently add a `PageReconstructError::MissingKey` for each key and your keyspace is sparse. I think the nicest option here is to simply not return anything for these keys and tweak the cross validation. Since we remove them from the keyspace, you don't need the `is_metadata_scan` check on line 3345.

Reply:
Sounds good. I added a sparse_non_inherited keyspace and discard missing key errors for it. https://github.com/neondatabase/neon/pull/7468/commits/91bba589efa2d34a66e0ae5724f775e65ad4ebf0